### PR TITLE
test(rule_engine): add update resource unit test case

### PR DESCRIPTION
**Pull request by wwhai**

### CODE REVIEW

File: apps/emqx_rule_engine/test/emqx_rule_engine_SUITE.erl [hunk 1]
```diff
@@ -447,8 +447,17 @@ t_crud_resources_api(_Config) ->
     {ok, #{code := 0, data := Resources2}} = emqx_rule_engine_api:show_resource(#{id => ResId},[]),
     ?assertEqual(ResId, maps:get(id, Resources2)),
 
-    ?assertMatch({ok, #{code := 0}}, emqx_rule_engine_api:delete_resource(#{id => ResId},#{})),
+    {ok, #{code := 0}} = emqx_rule_engine_api:update_resource(#{id => ResId},
+                                                           [{<<"id">>, ResId},
```
Comment (zhanghongtong) [inline apps/emqx_rule_engine/test/emqx_rule_engine_SUITE.erl:451]: Code indentation is in the wrong format

File: apps/emqx_rule_engine/src/emqx_rule_engine.erl [hunk 2]
```diff
@@ -309,6 +310,20 @@ delete_resource(ResId) ->
             {error, {resource_not_found, ResId}}
     end.
 
+update_resource(NewResource) ->
```
Comment (zhanghongtong) [inline apps/emqx_rule_engine/src/emqx_rule_engine.erl:313]:
```suggestion
update_resource(NewResource = #{id := Id,
                                config := Config,
                                type := Type,
                                description := Description,
                                created_at := CreatedAt}) ->
```

Comment (terry-xiaoyu) [reply apps/emqx_rule_engine/src/emqx_rule_engine.erl]: fixed

File: apps/emqx_rule_engine/test/emqx_rule_engine_SUITE.erl (hunk 1 above)
Comment (terry-xiaoyu) [reply apps/emqx_rule_engine/test/emqx_rule_engine_SUITE.erl]: fixed

**Pull request merged**